This patch fixes the "<command-line>:0:0: error: "__OBJECT_FILE__" redefined [-Werror]"
error when compiling Xen 4.7.0 on Fedora 23.

Needs also to change this in xen.spec:
-export CFLAGS="$RPM_OPT_FLAGS"
+export CFLAGS_EXTRA="$RPM_OPT_FLAGS"

See discussion here:
https://lists.xen.org/archives/html/xen-devel/2016-05/msg01277.html

Signed-off-by: Marcus of Wetware Labs <marcus@wetwa.re>
--- xen-4.7.0/config/StdGNU.mk  2016-06-20 13:38:15.000000000 +0300
+++ xen-4.7.0-new/config/StdGNU.mk      2016-07-19 16:08:35.886000000 +0300
@@ -36,7 +36,13 @@
 SHLIB_LDFLAGS = -shared

 ifneq ($(debug),y)
-CFLAGS += -O2 -fomit-frame-pointer
+CFLAGS += -O2 -fomit-frame-pointer -fno-tree-coalesce-vars
+  ifeq ($(XEN_TARGET_ARCH),x86_64)
+    #might be cross-compiling so strip out possible x86_32 options
+    CFLAGS += $(shell echo $(CFLAGS_EXTRA) | sed -e 's/-m32//g' -e 's/-march=i686//g' -e 's/-mtune=atom//g')
+  else
+    CFLAGS += $(CFLAGS_EXTRA)
+  endif
 else
 # Less than -O1 produces bad code and large stack frames
 CFLAGS += -O1 -fno-omit-frame-pointer

